textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcher.java

```diff
@@ -56,44 +62,83 @@ public void open() throws DataFlowException {
     @Override
     public ITuple getNextTuple() throws DataFlowException {
 		try {
-		    ITuple sourceTuple = inputOperator.getNextTuple();
-		    if (sourceTuple == null || !this.predicate.getIsSpanInformationAdded())
-		        return sourceTuple;
-            
-		    int schemaIndex = sourceTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST_ATTRIBUTE.getFieldName());
-		    List<Span> resultSpans =
-                    (List<Span>)sourceTuple.getField(schemaIndex).getValue();
-            
-		    /*The source operator returns spans even for those fields which did not satisfy the threshold criterion.
-		     *  So if two attributes A,B have 10 and 5 matching tokens, and we set threshold to 10,
-		     *  the number of spans returned is 15. So we need to filter those 5 spans for attribute B.
-		    */
-		    for(int attributeIndex = 0; attributeIndex < attributeList.size(); attributeIndex++) {
-		        String fieldName = attributeList.get(attributeIndex).getFieldName();
-		        IField field = sourceTuple.getField(fieldName);
-                
-		        if (field instanceof TextField) {         //Lucene defines Fuzzy Token Matching only for text fields.
-		            int tokensMatched = 0;
-		            List<Span> attributeSpans = new ArrayList<>();
-		            for (Span span : resultSpans) {
-		                if (span.getFieldName().equals(fieldName)) {
-		                    attributeSpans.add(span);
-		                    if (queryTokens.contains(span.getKey()))
-		                        tokensMatched++;
-		                }
-		            }
-		            if (tokensMatched < threshold) {
-		                resultSpans.removeAll(attributeSpans);
-		            }
-		        }
-		    }
-		    return sourceTuple;
+			if (cursor >= limit + offset - 1){
+				return null;
+			}
+			ITuple result = null;
+			while (true) {
+			    ITuple sourceTuple = inputOperator.getNextTuple();
```

Comment:
Again, use a function "computeNextTuple" to encapsulate the logic of computing the next valid tuple.  Then the main "while" loop should be the same for all operators.